### ISSUE TITLE
[member-ordering] add support for the new `accessor` syntax

### ISSUE THREAD

**Issue by vilicvane** [typescript-eslint/typescript-eslint#7903]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.2.2&fileType=.tsx&code=MYGwhgzhAEBiD29oG8CwAoa0zGAUyngCdoAjMIgLmgDsBXAW1LyIG4MMsAHASzAAoAlCmgBfDOPRA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AW0XYCNFpaA9tAAmfZgHN0UPtCGRwYAL4hFQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```typescript
class Foo {
  accessor bar: number;

  pia() { }
}
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/member-ordering": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

No error.

### Actual Result

Member pia should be declared before all method definitions. 4:3 - 4:12

**Comment by bradzacher** [comment]
The rule doesn't support the new accessor syntax so I think it's just being pushed to the bottom of the class because the rule doesn't know about the new AST node.

**Comment by vilicvane** [comment]
I guess so, PR on its way.
